### ISSUE TITLE
Primary key not defined for field '_id' in existing Realm file. Add @PrimaryKey.

### ISSUE THREAD

**Issue by LiuDeng** [realm/realm-java#1775]
This is my RealmConfiguration：
stableCongifuration = new RealmConfiguration.Builder(context).deleteRealmIfMigrationNeeded().name("stable.realm").build();
 
when id call Realm.getInstance(stableCongifuration); I got this exception:

11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:  Caused by: io.realm.exceptions.RealmMigrationNeededException: Primary key not defined for field '_id' in existing Realm file. Add @PrimaryKey.
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at io.realm.JokeRealmProxy.validateTable(JokeRealmProxy.java:443)
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at io.realm.DefaultRealmModuleMediator.validateTable(DefaultRealmModuleMediator.java:78)
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at io.realm.Realm.initializeRealm(Realm.java:342)
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at io.realm.Realm.createAndValidate(Realm.java:310)
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at io.realm.Realm.create(Realm.java:253)
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at io.realm.Realm.getInstance(Realm.java:217)
11-14 06:27:18.829 24386-24386/? E/AndroidRuntime:     at com.qiuduanribao.joke.utils.RealmUtils.getStableRealm(RealmUtils.java:32)

and the "_id" field have add @PrimaryKey

```
public class User extends RealmObject {
    @PrimaryKey
    private Long _id;
    private RealmList<Auth> auths;
    private String token;
    private String name;
    private boolean isMan;
    private String avatar;
}
```

Why ? How can i fix this ?

**Comment by cmelchior** [comment]
The error `JokeRealmProxy.validateTable(JokeRealmProxy.java:443)` points to a class called `Joke` not `User`. So presumably the error is there. I am still surprised it is throwing that error since you have `deleteRealmIfMigrationNeeded()` defined.

**Comment by LiuDeng** [comment]
@cmelchior loke at my Joke 
public class Joke extends RealmObject{
    private User U;
    @Index
    private Date T;
    private String text;
    private int shareC;
    @PrimaryKey
    private Long _id;

**Comment by LiuDeng** [comment]
@cmelchior I change my version to 0.84.1 and everything all right so it is a 0.84.2 bug

**Comment by cmelchior** [comment]
@LiuDeng Is this also happening on a fresh new install?

**Comment by LiuDeng** [comment]
@cmelchior yes  it is.

**Comment by LiuDeng** [comment]
@cmelchior  you did not update the .so file in 0.84.2 right? i use gradle to update realm to 0.84.2 but the .so file is still 0.84.1

**Comment by crosswall** [comment]
I have the same problem，When I updated realm to 0.84.2

**Comment by CherryPerry** [comment]
I have the same problem too after upgrading 0.84.1 to 0.84.2 via maven. All my realm object classes requires now realm migration to add primary index property.

**Comment by cmelchior** [comment]
@LiuDeng I just tried in a new sample project to create some data on 0.84.1 and updating to 0.84.2 without being able to reproduce the error. This exact upgrade path should also be covered by a unit test already: https://github.com/realm/realm-java/pull/1716/files#diff-3c101fdced0591f82692059985ee4b34R406

Are you able to reproduce this in a small sample project?

**Comment by beeender** [comment]
@LiuDeng Normally so file is updated for every release!! So please update the the `so` file when you update Realm!

**Comment by beeender** [comment]
@LiuDeng We can reproduce this issue now, and we are trying to fix it now.

**Comment by skyisle** [comment]
I met same exception. I tried to upgarde 0.84.1 to 0.85.0.

**Comment by skyisle** [comment]
#1775 is related to this issue.

**Comment by beeender** [comment]
Fix has been merged and will be release in 0.85.1 .

**Comment by jiajiaojie** [comment]
Uninstall the application first and reinstall it. It will work.
